docs/tutorials/js-sdk-tutorial.md

```diff
@@ -0,0 +1,144 @@
+---
+
+## Introduction
+
+The [DataHub](http://datahub.io/) platform stores a lot of different datasets - which are packages of useful data alongside with the description (here is [the dataset specification](https://frictionlessdata.io/docs/data-package/)). The data, stored on the DataHub, has a nice structure, views and a description that help people to get insights.
+You can also store and share your own datasets on the DataHub
+
+As a programmer, you may want to automate the process of getting or storing the data. Also you may want to integrate your project and the DataHub.
+
+This tutorial will cover information you need to create the data driven project that will use the DataHub to get or store the data.
+
+Let's explore the tutorial together below.
+
+**Important notes:**
+- You need to use **Node version > 7.6**
+- When you see the **await** keyword you should wrap the code in the **async** function.
+
+
+## Install
+
+Add `datahub-client` package into your project:
+```shell
+npm install datahub-client --save
+```
+The `datahub-client` is a JS library to communicate with the DataHub and to create and validate datasets. It also includes the `data.js` package.
+`Data.js` provides `Dataset` and `File` classes, that represents a datapackage and a data-file.
+
+
+## Quick overview
+
+After loading 'datahub-client' into your program:
+`const datahub = require('datahub-client')` ,
+you can do things like:
+
+- login to DataHub
+- authenticate with the jwt token
+- push a dataset:
+- get the data from the DataHub:
+- init a dataset
+- validate a dataset
+- convert tabular data files into different formats
+
+Let's explore some of the `datahub-client` features more deeply below.
+
+
+## Push data
+
+### Prepare the test data and user credentials
+1. Here is a [Sample dataset on Github](https://github.com/datasets/finance-vix/archive/master.zip).  Please, download and unpack it.
+
+2. Get the user credentials from the server (token and ownerid):
+   - install our cli tool: [installation instructions](https://datahub.io/docs/getting-started/installing-data)
+   - run `data login` and login on the DataHub in the browser
+   - run `cat ~/.config/datahub/config.json`
+```
+{
+  "token": "secure jwt token",
+  "profile": {
+    "id": "userId",
+    /* other user info */
+  }
+}
+```
+
+### Push code example
+
+```javascript
+const {DataHub} = require('datahub-client')
+const {Dataset} = require('data.js')
+
+async function pushDataset(datasetPath){
+   // load dataset that we want to push
+   let dataset = await Dataset.load(datasetPath)
+   // Create an instance of the DataHub class, using the data from the user config
+   const datahubConfigs = {
+     apiUrl: 'http://api.datahub.io/',
+     token: 'secure jwt token',
+     debug: false,
+     ownerid: 'userId',
+   }
+   const datahub = new DataHub(datahubConfigs)
+
+   // now use the datahub instance to push the data
+   const res = await datahub.push(dataset, {findability: 'unlisted'})
```

Comment:
Let's make `published` this one.

Reply:
fixed (btw why it is not called `PUBLIC`, think it is much better then published ?)

Reply:
agree, but for now that's so